### PR TITLE
Add support for permanova

### PR DESCRIPTION
**Pull request by sfiligoi**
ssu can now compute PERMANOVA.
Needs access to a TSV file.

The data is stored in the hdf5 file as a list of stats elements.

### CODE REVIEW

**Comment by sfiligoi** [comment]
Addresses part of #30

**Review by wasade — COMMENTED** [review]
Parsing TSV in c++ does not look fun...

Is there any chance a some unit tests mirroring the skbio PERMANOVA tests, and additionally some tests on the TSV parsing? Former are verified against the original implementation and the latter is important as edge cases could effect the statistics

**Comment by sfiligoi** [comment]
> Parsing TSV in c++ does not look fun...
> 
> Is there any chance a some unit tests mirroring the skbio PERMANOVA tests, and additionally some tests on the TSV parsing? Former are verified against the original implementation and the latter is important as edge cases could effect the statistics

Yes, we should add some unit tests.
The big picture tests (against python version) are part of the github tests.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -143,7 +143,18 @@ jobs:
         ./compare_unifrac_pcoa.py test500.unweighted_fp32.f.h5 t1.h5 3 0.1
         ls -l t1.h5
         rm -f t1.h5
+        # permanova
+        echo "permanova"
+        time ssu -m unweighted -i test500.biom  -t test500.tre --pcoa 4  -r hdf5 --permanova 127 -g test500.tsv -c empo_2 -o t1.h5
+        python compare_unifrac_stats.py t1.h5 5 127 1.0 0.484 0.01
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply .github/workflows/main.yml:149]: Yes, the test values came from the skbio version.

File: src/su.cpp [hunk 2]
```diff
@@ -536,17 +582,33 @@ int main(int argc, char **argv){
     else
         pcoa_dims = atoi(pcoa_arg.c_str());
 
+    unsigned int permanova_perms;
+    if(permanova_arg.empty()) {
+        if (grouping_filename.empty() || grouping_columns.empty()) {
+            // cannot compute permanova without the grouping file and columns
+            permanova_perms = 0;
+        } else {
+            permanova_perms = 1023;
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply src/su.cpp:591]: 1023 would make more sense, compute wise. Power of 2 are more effective.

File: README.md [hunk 3]
```diff
@@ -158,6 +160,7 @@ The methods can be used directly through the command line after install:
                                  hdf5_fp32 : HFD5 format, using fp32 precision.
                                  hdf5_fp64 : HFD5 format, using fp64 precision.
                                  hdf5_nodist : HFD5 format, no distance matrix, just PCoA.
+        --permanova	[OPTIONAL] Number of Permanova permutations to compute (default: 1023 with -g, do not compute if 0)
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply README.md]: I guess it makes sense to mimic historic precedents.
